samples/dnn/dasiamrpn_tracker.cpp

```diff
@@ -0,0 +1,468 @@
+{
+    Mat pad = (w + h)*0.5;
+    Mat sz2 = (w + pad).mul((h + pad));
+
+    cv::sqrt(sz2, sz2);
+    return sz2;
+}
+
+int main(int argc, char **argv)
+{
+    // Parse command line arguments.
+    CommandLineParser parser(argc, argv, keys);
+
+   if (argc == 1 || parser.has("help"))
+   {
+       parser.printMessage();
+       return 0;
+   }
+
+    std::string inputName = parser.get<String>("input");
+    std::string net = parser.get<String>("net");
+    std::string kernel_cls1 = parser.get<String>("kernel_cls1");
+    std::string kernel_r1 = parser.get<String>("kernel_r1");
+
+    // Read nets.
+    Net siamRPN, siamKernelCL1, siamKernelR1;
+    try
+    {
+        siamRPN = readNet(samples::findFile(net));
+        siamKernelCL1 = readNet(samples::findFile(kernel_cls1));
+        siamKernelR1 = readNet(samples::findFile(kernel_r1));
+    }
+    catch (const cv::Exception& ee)
+    {
+        std::cerr << "Exception: " << ee.what() << std::endl;
+        std::cout << "Can't load the network by using the flowing files:" << std::endl;
+        std::cout << "siamRPN : " << net << std::endl;
+        std::cout << "siamKernelCL1 : " << kernel_cls1 << std::endl;
+        std::cout << "siamKernelR1 : " << kernel_r1 << std::endl;
+        return 1;
+    }
+
+    const std::string preWinName = "DaSiamRPN";
+    namedWindow(preWinName, WINDOW_AUTOSIZE);
+
+    // Open a video file or an image file or a camera stream.
+    VideoCapture cap;
+    Mat image;
+
+    if( inputName.empty() || (isdigit(inputName[0]) && inputName.size() == 1) )
+    {
+        int c = inputName.empty() ? 0 : inputName[0] - '0' ;
+        if(!cap.open(c))
+        {
+            std::cout << "Capture from camera #" <<  c << " didn't work" << std::endl;
+            return -1;
+        }
+    }
+    else if( inputName.size() )
+    {
+        inputName = samples::findFileOrKeep(inputName);
+        if(!cap.open( inputName ))
+        {
+            std::cout << "Could not read " << inputName << std::endl;
+            return -1;
+        }
+    }
+
+    // Read the first image.
+    cap >> image;
+    if (image.empty())
+    {
+        std::cout << "Image reading error!"<< std::endl;
+        return -1;
+    }
+
+    putText(image, "Select initial bounding box you want to track.", Point(0, 15), FONT_HERSHEY_SIMPLEX, 0.5, Scalar(0, 255, 0));
+    putText(image, "And Press the ENTER key.", Point(0, 35), FONT_HERSHEY_SIMPLEX, 0.5, Scalar(0, 255, 0));
+
+    trackerConfig trackState;
+    trackState.update_scoreSize();
+
+    Rect selectRect = selectROI(preWinName, image);
+
+    trackState.targetBox = Rect2f{float(selectRect.x) + float(selectRect.width)/2, float(selectRect.y) + float(selectRect.height)/2, float(selectRect.width), float(selectRect.height)};
+
+    // Set tracking template.
+    trackerInit(image, trackState, siamRPN, siamKernelR1, siamKernelCL1);
+
+    TickMeter tickMeter;
+
+    siamRPN.setPreferableBackend(5);
+    siamRPN.setPreferableTarget(DNN_TARGET_CUDA);
```

Comment:
> 5

Avoid magic numbers.

> DNN_TARGET_CUDA

Please make this as sample command-line parameter. See [here](https://github.com/opencv/opencv/blob/4.5.1/samples/dnn/classification.cpp#L17-L26)

---

BTW, There are multiple networks used.

Reply:
Thanks, I will remove these code for setting the dnn backend and let him choose it automatically.

Reply:
Thanks, fixed.